core/kogitobuild/build_config.go

```diff
@@ -17,245 +17,41 @@ package kogitobuild
-		fromImage := getBuilderImageStreamOutputTo(build)
-		bc.Spec.Source.Type = buildv1.BuildSourceImage
-		bc.Spec.Source.Images = []buildv1.ImageSource{
-			{
-				From: *fromImage,
-				Paths: []buildv1.ImageSourcePath{
-					{
-						DestinationDir: destinationDir,
-						SourcePath:     runnerSourcePath,
-					},
-				},
-			},
-		}
-		// triggers once we have a change in the s2i produced image (builder BuildConfig)
-		bc.Spec.Triggers = []buildv1.BuildTriggerPolicy{
-			{
-				Type:        buildv1.ImageChangeBuildTriggerType,
-				ImageChange: &buildv1.ImageChangeTrigger{From: fromImage},
-			},
-		}
-	}
-}
-
-// decoratorForRuntimeBuilder decorates the original BuildConfig to give very basic support for any Runtime builders, the
-// ones responsible to build the final KogitoRuntime image.
-// should be used with `decoratorForSourceRuntimeBuilder` or `decoratorForBinaryRuntimeBuilder` to add full support for these use cases.
-func (b *decoratorHandler) decoratorForRuntimeBuilder() decorator {
-	return func(build api.KogitoBuildInterface, bc *buildv1.BuildConfig) {
-		bc.Name = build.GetName()
-		imageStreamHandler := NewImageSteamHandler(b.Context)
-		baseImage := corev1.ObjectReference{
-			Kind:      kindImageStreamTag,
-			Namespace: build.GetNamespace(),
-			Name:      imageStreamHandler.ResolveKogitoImageStreamTagName(build, false),
-		}
-		// this image stream will be the input for the KogitoRuntime deployment.
-		bc.Spec.Output.To = &corev1.ObjectReference{
-			Kind: kindImageStreamTag, Name: strings.Join([]string{GetApplicationName(build), tagLatest}, ":"),
-		}
-
-		bc.Spec.Strategy = buildv1.BuildStrategy{
-			Type: buildv1.SourceBuildStrategyType,
-			SourceStrategy: &buildv1.SourceBuildStrategy{
-				From: baseImage,
-				Env:  build.GetSpec().GetEnv(),
-			},
-		}
-	}
-}
-
-// getBuilderImageStreamOutputTo gets the builder ImageStream which is the one that we will output to when building from source
-// and the one we will use as input when building the final image.
-func getBuilderImageStreamOutputTo(build api.KogitoBuildInterface) *corev1.ObjectReference {
-	return &corev1.ObjectReference{
-		Kind: kindImageStreamTag, Name: strings.Join([]string{GetBuildBuilderName(build), tagLatest}, ":"),
-	}
-}
-
-// newBuildConfig creates a new reference for the very basic default OpenShift BuildConfig reference to build Kogito Services.
+// NewBuildConfig creates a new reference for the very basic default OpenShift BuildConfig reference to build Kogito Services.
 // Pass the required decorator(s) to create the BuildConfig for a particular use case
-func newBuildConfig(build api.KogitoBuildInterface, decorators ...decorator) buildv1.BuildConfig {
-	app := build.GetSpec().GetTargetKogitoRuntime()
-	if len(app) == 0 {
-		app = build.GetName()
-	}
+func (b *buildConfigHandler) NewBuildConfig(build api.KogitoBuildInterface, customLabel map[string]string, decorators ...decorator) buildv1.BuildConfig {
```

Comment:
Why do we need to expose this function? I didn't see a new usage outside of this package.

Reply:
```suggestion
func (b *buildConfigHandler) NewBuildConfig(build api.KogitoBuildInterface, customLabels map[string]string, decorators ...decorator) buildv1.BuildConfig {
```

Reply:
done